homeassistant/components/hue/sensor_base.py

```diff
@@ -0,0 +1,229 @@
+"""Support for the Philips Hue sensors as a platform."""
+import asyncio
+from datetime import timedelta
+import logging
+from time import monotonic
+
+import async_timeout
+
+from homeassistant.components import hue
+from homeassistant.helpers.event import async_track_point_in_utc_time
+from homeassistant.util.dt import utcnow
+
+
+SCAN_INTERVAL = timedelta(seconds=5)
+CURRENT_SENSORS = 'current_sensors'
+
+PRESENCE_NAME_FORMAT = "{} presence"
+LIGHT_LEVEL_NAME_FORMAT = "{} light level"
+TEMPERATURE_NAME_FORMAT = "{} temperature"
+
+
+_HUE_SENSOR_TYPE_CONFIG_MAP = {}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _device_id(aiohue_sensor):
+    # Work out the shared device ID, as described below
+    device_id = aiohue_sensor.uniqueid
+    if device_id and len(device_id) > 23:
+        device_id = device_id[:23]
+    return device_id
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities,
+                            binary=False):
+    """Set up the Hue sensors from a config entry."""
+    import aiohue
+    from homeassistant.components.hue.binary_sensor import HuePresence
+    from homeassistant.components.hue.sensor import (
+        HueLightLevel, HueTemperature)
+
+    _HUE_SENSOR_TYPE_CONFIG_MAP.update({
+        aiohue.sensors.TYPE_ZLL_LIGHTLEVEL: {
+            "binary": False,
+            "name_format": LIGHT_LEVEL_NAME_FORMAT,
+            "class": HueLightLevel,
+        },
+        aiohue.sensors.TYPE_ZLL_TEMPERATURE: {
+            "binary": False,
+            "name_format": TEMPERATURE_NAME_FORMAT,
+            "class": HueTemperature,
+        },
+        aiohue.sensors.TYPE_ZLL_PRESENCE: {
+            "binary": True,
+            "name_format": PRESENCE_NAME_FORMAT,
+            "class": HuePresence,
+        },
+    })
+
+    bridge = hass.data[hue.DOMAIN][config_entry.data['host']]
+    hass.data[hue.DOMAIN][CURRENT_SENSORS] = {}
+
+    async def async_update_bridge(now):
```

Comment:
Wouldn't this hammer the sensor endpoint twice?

Reply:
Resolved in the latest refactor.